test/uk/gov/hmrc/cgtpropertydisposalsfrontend/connectors/upscan/UpscanConnectorImplSpec.scala

```diff
@@ -31,46 +34,72 @@ import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.generators.FileUploadGen.
 import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.generators.IdGen._
 import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.upscan.{UploadReference, UpscanInitiateRequest, UpscanUpload}
-import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse}
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.util.WireMockMethods
+import uk.gov.hmrc.http.test.WireMockSupport
+import uk.gov.hmrc.http.{HeaderCarrier, HttpClient, HttpResponse}
 import uk.gov.hmrc.play.bootstrap.config.ServicesConfig
 
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 
-class UpscanConnectorImplSpec extends AnyWordSpec with Matchers with MockFactory with HttpSupport {
+class UpscanConnectorImplSpec
+  extends AnyWordSpec
+    with Matchers
+    with MockFactory
+    with GuiceOneAppPerSuite
+    with WireMockSupport
+    with WireMockMethods
+    with EitherValues with HttpSupport {
 
   private val config = Configuration(
     ConfigFactory.parseString(
-      """
+      s"""
         | self {
         |   url = host1.com
         |  },
         |  microservice {
         |    services {
-        |      upscan-initiate {
-        |        protocol = https
-        |        host     = host2
-        |        port     = 123
-        |        user-agent = agent
-        |        max-file-size = 1234
-        |      },
         |      cgt-property-disposals {
-        |        protocol = https
-        |        host     = host3
-        |        port     = 123
+        |        protocol = http
+        |        host     = $wireMockHost
+        |        port     = $wireMockPort
         |      }
         |   }
         |}
         |""".stripMargin
     )
   )
 
-  val connector =
-    new UpscanConnectorImpl(
-      mockHttp,
-      config,
-      new ServicesConfig(config)
+  private val initiateConfig = Configuration(
+    ConfigFactory.parseString(
+      s"""
+         | self {
+         |   url = host1.com
+         |  },
+         |  microservice {
+         |    services {
+         |      upscan-initiate {
+         |        protocol = http
+         |        host     = $wireMockHost
+         |        port     = $wireMockPort
+         |        user-agent = agent
+         |        max-file-size = 1234
+         |      },
+         |     cgt-property-disposals {
+         |        protocol = https
+         |        host     = host3
+         |        port     = 123
+         |      }
+         |   }
+         |}
+         |""".stripMargin
     )
+  )
+
+  override def fakeApplication(): Application = new GuiceApplicationBuilder().configure(config).build()
+  val httpClient: HttpClient = app.injector.instanceOf[HttpClient]
+  val connector = new UpscanConnectorImpl(httpClient, config = config, servicesConfig = new ServicesConfig(config))
+  val upScanConnector = new UpscanConnectorImpl(httpClient, config = initiateConfig, servicesConfig = new ServicesConfig(initiateConfig))
```

Comment:
Once the config is combined, above comment, we should resolve back to a single connector rather than 2 instances here.
Also, we won't need to inject `HttpClient` and `ServicesConfig` explicitly here. Once you the fakeApplication available using the config, you could just do following and everything (`HttpClient`, `Config` & `ServicesConfig`) should get wired into UpscanConnectorImpl accordingly. So, it would just be 

`val connector = app.injector.instanceOf[UpscanConnectorImpl]`